### PR TITLE
Add pytest-flake8

### PR DESCRIPTION
**Pull request by noviluni**
proposal for: https://github.com/scrapy/scrapy/issues/3944

~Please, take in mind that it’s only a PoC. In this way the testing time increases and maybe it’s not necessary to check flake8 against all environments.~

### CODE REVIEW

**Comment by noviluni** [comment]
Seems that pipeline worked: https://travis-ci.org/scrapy/scrapy/jobs/569955789

**Comment by Gallaecio** [comment]
@noviluni Looks great. However be ready for something like https://github.com/scrapy/scrapy/pull/3727#issuecomment-501902135 from @dangra :)

**Comment by noviluni** [comment]
Hi @Gallaecio thank you for your feedback. I've been thinking that, and I'm not totally agree (in fact, I can't fully understand the reasons)... The idea here is to avoid big PRs to be reviewed and adding `noqa` marks to every line/function/file implies a lot of changes :/

As my original proposal was to avoid that, for me it has no sense. I can't see the benefits of adding all those lines and then delete them when fixing the code. It involves more work (twice) and probably more "risk" (looking at your PR: 170 files to review is kind of risky).

@dangra what do you think? Is it necessary to mark all the cases one by one and then start fixing them or could we start this way and when fixing every flake8 rule add the necessary `noqa` to those "exceptions" we want to keep?

Thanks in advance

**Comment by noviluni** [comment]
Updated to use a list files (with the excluded rules) instead of ignoring all rules directly. In that way is possible to disable checks only for the files that are not passing them.

File: .travis.yml [hunk 1]
```diff
@@ -7,6 +7,8 @@ branches:
     - /^\d\.\d+\.\d+(rc\d+|\.dev\d+)?$/
 matrix:
   include:
+    - env: TOXENV=flake8
+      python: 3.5
```
Comment (Gallaecio) [inline .travis.yml]: Shouldn’t we use 3.7? (also relevant for `tox.ini`)

File: pytest.ini [hunk 2]
```diff
@@ -4,3 +4,256 @@ python_files=test_*.py __init__.py
...
+    scrapy/pqueues.py E501
+    scrapy/resolver.py E302
+    scrapy/responsetypes.py E128 E501 E305
+    scrapy/robotstxt.py E302 E501
+    scrapy/shell.py E501
+    scrapy/signalmanager.py E501
+    scrapy/spiderloader.py E225 F841 E501 E126
+    scrapy/squeues.py E128
+    scrapy/statscollectors.py E501 W391
+    # tests
```
Comment (Gallaecio) [inline pytest.ini]: Shouldn’t tests be excluded from these checks?

File: .travis.yml (hunk 1 above)
Comment (noviluni) [reply .travis.yml]: It's ok for me, in fact I don't remember why I put 3.5.... Already pushed a commit with this change, thank you :+1:

File: pytest.ini (hunk 2 above)
Comment (noviluni) [reply pytest.ini]: hmmm I'm don't see any reason to do that, could you explain me why? :thinking: 

From my point of view, fixing the PEP8/style in the tests will contribute to their maintainability.

**Comment by noviluni** [comment]
> I’ve left a couple of comments, but I’m OK with merging this _as is_.

thank you for your feedback

**Pull request merged**